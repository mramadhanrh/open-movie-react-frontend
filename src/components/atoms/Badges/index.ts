import styled, { ColorVariant } from 'styled-components';
import defaultTheme from '../../../constants/theme';

interface BadgesProps {
  color?: ColorVariant;
  textColor?: ColorVariant;
}

const Badges = styled.div<BadgesProps>`
  background-color: ${({ color }) => defaultTheme.color[color || 'ecstasy']};
  border-radius: 3px;
  font-weight: 500;
  color: ${({ textColor }) => defaultTheme.color[textColor || 'white']};
  width: fit-content;
  padding: 5px 10px;
  font-size: 12px;
`;

export default Badges;
