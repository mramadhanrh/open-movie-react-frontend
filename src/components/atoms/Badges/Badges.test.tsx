import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Badges from '.';

describe('Badges', () => {
  test('match snapshot', () => {
    expect(toJson(shallow(<Badges />))).toMatchSnapshot();
  });

  test('render Text', () => {
    expect(shallow(<Badges>Hello</Badges>).text()).toBe('Hello');
  });
});
