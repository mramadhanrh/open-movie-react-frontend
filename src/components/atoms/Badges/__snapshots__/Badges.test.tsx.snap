// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badges match snapshot 1`] = `
<div
  className="sc-bdnxRM ksguFW"
/>
`;
